src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.cs

```diff
@@ -112,8 +113,18 @@ public NamedPipeServerStream(String pipeName, PipeDirection direction, int maxNu
                 throw new ArgumentOutOfRangeException(nameof(inheritability), SR.ArgumentOutOfRange_HandleInheritabilityNoneOrInheritable);
             }
 
+            if ((options & PipeOptions.CurrentUserOnly) != 0)
+            {
+                IsCurrentUserOnly = true;
+
+                // We need to remove this flag from options because it is not a valid flag for windows PInvoke to create a pipe.
+                options &= ~PipeOptions.CurrentUserOnly;
```

Comment:
This feels like it should happen in Windows specific code - like the Create method.  It just feel odd to remove this flag here.

Reply:
But then wouldn’t we have to duplicate code on both create methods?

Reply:
The comment only says "because it is not a valid flag for windows PInvoke to create a pipe."  It doesn't say that it is not a valid flag for Unix.  So I would see no reason why Unix would care if this flag is set or not.

Reply:
Yes, the only specific Windows piece of code here is when removing the flag. So I can go ahead and move the deletion of the flag over to the Create method in Windows.

Reply:
Also, for the client windows code we would have to remove it in TryConnect which is called multiple times if the pipe is busy while trying to connect. In the common code we would only be removing the flag once.